Automattic/jetpack-boost-critical-css-gen
# Refactor Critical CSS Generation logic

Linked issue: https://github.com/Automattic/jetpack/issues/20586

### Changes proposed in this Pull Request:
1. Move the logic to find matching selectors including above-the-fold on the page to a static method in BrowserInterface class.
2. Move the logic to get the above-the-fold selectors list to a separate function.
3. To better implement the progress callback, added a wrapper to run update progress callback.
4. Updated `npm install` to `npm ci` in Github actions for CI consistency.

Thanks, @thingalon for implementing the major part.

### Testing instructions:
Run `node bin/generate {url}` and verify the generated Critical CSS output.
**OR**
Prepare the bundle with `npm run bundle`
Use the bundled js in place of `app/assets/dist/critical-css-gen.js` in Boost plugin.
